src/me/drton/flightplot/export/AbstractTrackExporter.java

```diff
@@ -37,6 +38,9 @@ private void initAnalyzers() {
     protected TrackPoint readNextPoint() throws IOException, FormatErrorException {
         TrackPoint point = this.trackReader.readNextPoint();
         if (null != point) {
+            if (altOffset != 0) {
+                point = new TrackPoint(point.lat, point.lon, (float)(point.alt + altOffset), point.time);
```

Comment:
Because you're creating a new point here, the one (optional) parameter that is not in the constructor gets lost (`flightMode`). Just update the `alt` member of point. I left those public intentionally to make it a little easier to work with:
```
point.alt = point.alt + altOffset;
```